lib/phoenix_token_auth/registrator.ex

```diff
@@ -8,13 +8,23 @@ defmodule PhoenixTokenAuth.Registrator do
   Validates that email and password are present and that email is unique.
   """
   def changeset(params) do
-    Changeset.cast(struct(UserHelper.model), params, ~w(email))
-    |> Changeset.validate_change(:email, &Util.presence_validator/2)
-    |> Changeset.validate_unique(:email, on: Util.repo)
-    |> UserHelper.validator
-    |> set_hashed_password
-  end
 
+    if params["username"] do
+      Changeset.cast(struct(UserHelper.model), params, ~w(username))
+      |> Changeset.validate_change(:username, &Util.presence_validator/2)
+      |> Changeset.validate_unique(:username, on: Util.repo)
+      |> UserHelper.validator
+      |> set_hashed_password
+      |> Changeset.put_change(:hashed_confirmation_token, nil)
+      |> Changeset.put_change(:confirmed_at, Ecto.DateTime.utc)
+    else
+      Changeset.cast(struct(UserHelper.model), params, ~w(email))
+      |> Changeset.validate_change(:email, &Util.presence_validator/2)
+      |> Changeset.validate_unique(:email, on: Util.repo)
+      |> UserHelper.validator
+      |> set_hashed_password
+    end
+  end
```

Comment:
i would prefer allowing email and username attributes to be set at the same time here. (`~(username)` -> `~(email username)`)
then we need to run both uniqueness validations though.
then i would prefer to have the `changeset` method do the things both cases have in common and later call a private method that implements the logic that depends on whether we are in the username or email case. this method would match on the params again.